docs/src/main/tut/step-by-step-example.md

```diff
@@ -0,0 +1,199 @@
+## Example
+
+In this example we will build an api to store and retrieve users.
+
+### 1 - Model and Controller Interface
+
+First, let's start by defining the interface of the API controller. We want the API to support the two following
+operations:
+
+1. The client should be able to insert a new user by specifying `id` and `name`
+2. The client should be able to retrieve a user by `id`
+
+The following snippet defines a model for `User` and an interface that follows this specification.
+
+```scala
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+// Models definition
+object models {
+  case class User(name: String)
+}
+
+object controllers {
+  import models._
+  import wiro.annotation._
+
+  // Error messages
+  case class Error(msg: String)
+  case class UserNotFoundError(msg: String)
+
+  // API interface
+  @path("users")
+  trait UsersApi {
+
+    @query(name = Some("getUser"))
```

Comment:
since you're not renaming the method you can probably just do `@query` (same for `@command` below).
/cc @calippo, right?